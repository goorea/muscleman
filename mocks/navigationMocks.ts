export const navigationDispatchMock = jest.fn();
export const navigationNavigateMock = jest.fn();
export const navigationGoBackMock = jest.fn();
export const navigationSetOptionsMock = jest.fn();
export const navigationSetParamsMock = jest.fn();
export const navigationCanGoBackMock = jest.fn();
export const navigationGetParentMock = jest.fn();
export const navigationGetStateMock = jest.fn();
export const navigationReplaceMock = jest.fn();
export const navigationIsFocusedMock = jest.fn();
export const navigationResetMock = jest.fn();
export const navigationPopMock = jest.fn();
export const navigationPushMock = jest.fn();
export const navigationPopToTopMock = jest.fn();
export const navigationAddListenerMock = jest.fn();
export const navigationRemoveListenerMock = jest.fn();
export const navigationJumpToMock = jest.fn();
export const navigationMock = {
  dispatch: navigationDispatchMock,
  navigate: navigationNavigateMock,
  goBack: navigationGoBackMock,
  setOptions: navigationSetOptionsMock,
  setParams: navigationSetParamsMock,
  canGoBack: navigationCanGoBackMock,
  getParent: navigationGetParentMock,
  getState: navigationGetStateMock,
  isFocused: navigationIsFocusedMock,
  reset: navigationResetMock,
  addListener: navigationAddListenerMock,
  removeListener: navigationRemoveListenerMock,
  replace: navigationReplaceMock,
  pop: navigationPopMock,
  push: navigationPushMock,
  popToTop: navigationPopToTopMock,
  jumpTo: navigationJumpToMock,
};
