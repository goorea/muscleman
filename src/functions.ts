import StackFlashMessage, { Options } from 'react-native-stack-flash-message';

import { WeightSet } from '@src/types';
import { Set, TrainingType } from '@src/types/graphql';

export const flash = (options: Options) => StackFlashMessage.show(options);

export const getTrainingTypeForKorean = (type: TrainingType): string => {
  switch (type) {
    case TrainingType.Lower:
      return '하체';
    case TrainingType.Chest:
      return '가슴';
    case TrainingType.Back:
      return '등';
    case TrainingType.Shoulder:
      return '어깨';
    case TrainingType.Arm:
      return '팔';
    case TrainingType.Abdominal:
      return '복근';
    case TrainingType.Cardiovascular:
      return '유산소';
    default:
      return '기타';
  }
};

export const isWeightSets = (sets: Set[]): sets is WeightSet[] =>
  sets.every(
    set =>
      (set as WeightSet).weight !== undefined &&
      (set as WeightSet).count !== undefined,
  );
