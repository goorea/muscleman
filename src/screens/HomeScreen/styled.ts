import styled from 'styled-components/native';

import Button from '@src/components/Button';
import Image from '@src/components/Image';

export const Container = styled.ScrollView`
  padding: 20px;
`;

export const UserContainer = styled(Button)`
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
`;

export const ProfileImage = styled(Image)`
  width: 40px;
  height: 40px;
  border-radius: 20px;
  margin-right: 10px;
`;

export const PlansContainer = styled.View`
  margin-top: 40px;
`;

export const EmptyPlansContainer = styled.View`
  margin-top: 10px;
  flex-direction: row;
  align-items: center;
`;

export const Notice = styled(Button)`
  width: 26px;
  height: 26px;
  border-radius: 13px;
  padding: 0;
  margin-right: 10px;
`;

export const ToPlan = styled(Button)`
  border-radius: 8px;
  margin-left: 10px;
  padding-left: 0;
  padding-right: 0;
  flex: 1;
`;
