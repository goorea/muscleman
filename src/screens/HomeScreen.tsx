import React from 'react';
import { MainTabParamList, RootStackParamList } from '@src/types/navigation';
import { BottomTabScreenProps } from '@react-navigation/bottom-tabs';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { CompositeScreenProps } from '@react-navigation/native';
import { Button } from 'react-native-elements';
import styled from 'styled-components/native';
import { flexFillCenter } from '@src/styles/flex';
import Text from '@src/components/Text';

const Container = styled.View`
  ${flexFillCenter}
`;

type P = CompositeScreenProps<
  BottomTabScreenProps<MainTabParamList, 'Home'>,
  NativeStackScreenProps<RootStackParamList>
> & {};

const HomeScreen: React.FC<P> = ({ navigation }) => (
  <Container>
    <Text>Home Screen</Text>
    <Button
      title="Login"
      onPress={() => navigation.navigate('Auth', { screen: 'Login' })}
    />
  </Container>
);

export default HomeScreen;
