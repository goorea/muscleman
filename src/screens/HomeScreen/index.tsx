import { BottomTabScreenProps } from '@react-navigation/bottom-tabs';
import { CompositeScreenProps } from '@react-navigation/native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import React from 'react';

import Button from '@src/components/Button';
import Text from '@src/components/Text';
import { MainTabParamList, RootStackParamList } from '@src/types/navigation';

import { Container } from './styled';

type P = CompositeScreenProps<
  BottomTabScreenProps<MainTabParamList, 'Home'>,
  NativeStackScreenProps<RootStackParamList>
>;

const HomeScreen: React.FC<P> = ({ navigation }) => (
  <Container>
    <Text>Home Screen</Text>
    <Button
      title="Login"
      onPress={() => navigation.navigate('Auth', { screen: 'Login' })}
    />
  </Container>
);

export default HomeScreen;
