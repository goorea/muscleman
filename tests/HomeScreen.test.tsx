import { navigationMock } from '@mocks/navigationMocks';
import { render } from '@testing-library/react-native';
import React from 'react';

import HomeScreen from '@src/screens/HomeScreen';
import { wrapper } from '@tests/functions';

describe('HomeScreen 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    const { toJSON } = render(
      <HomeScreen
        navigation={navigationMock}
        route={{
          key: '',
          name: 'Home',
        }}
      />,
      { wrapper },
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
