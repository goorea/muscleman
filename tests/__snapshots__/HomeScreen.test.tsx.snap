// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen 컴포넌트 렌더링이 올바르게 된다 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    />
    <RCTScrollView
      activeAnimationOptions={null}
      activeAnimationType="timing"
      activeSlideAlignment="center"
      activeSlideOffset={20}
      apparitionDelay={0}
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDelay={1000}
      autoplayInterval={3000}
      callbackOffsetMargin={5}
      containerCustomStyle={Object {}}
      contentContainerCustomStyle={Object {}}
      contentContainerStyle={
        Array [
          Object {
            "paddingLeft": 0,
            "paddingRight": 0,
          },
          Object {},
        ]
      }
      data={
        Array [
          <Styled(LinearGradient)
            colors={
              Array [
                "#2687ED",
                "#ca71eb",
              ]
            }
            end={
              Object {
                "x": 1,
                "y": 1,
              }
            }
            start={
              Object {
                "x": 0,
                "y": 0,
              }
            }
          >
            <Styled(View)>
              <View>
                <Memo(Text)
                  color="white"
                  weight="bold"
                >
                  나의 3대 측정
                </Memo(Text)>
                <Styled(View)>
                  <Memo(Icon)
                    color="white"
                    name="calendar-today"
                    size={16}
                  />
                  <Styled(Component)
                    color="white"
                    size={12}
                  >
                    21\`11.16
                  </Styled(Component)>
                </Styled(View)>
              </View>
            </Styled(View)>
            <Styled(View)>
              <Memo(Text)
                color="white"
                weight="bold"
              >
                로그인 한 후에 확인할 수 있어요!
              </Memo(Text)>
              <Styled(Component)
                color="white"
                onPress={[Function]}
                size={12}
                title="로그인 하러가기"
                type="outline"
              />
            </Styled(View)>
          </Styled(LinearGradient)>,
        ]
      }
      decelerationRate="fast"
      directionalLockEnabled={true}
      enableMomentum={false}
      enableSnap={true}
      firstItem={0}
      forwardedRef={[Function]}
      getItem={[Function]}
      getItemCount={[Function]}
      hasParallaxImages={false}
      horizontal={true}
      inactiveSlideOpacity={0.7}
      inactiveSlideScale={0.9}
      inactiveSlideShift={0}
      initialNumToRender={6}
      invertStickyHeaders={false}
      inverted={false}
      itemWidth={710}
      keyExtractor={[Function]}
      layout="default"
      lockScrollTimeoutDuration={1000}
      lockScrollWhileSnapping={false}
      loop={false}
      loopClonesPerSide={3}
      maxToRenderPerBatch={13}
      onBeforeSnapToItem={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onResponderRelease={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      overScrollMode="never"
      pinchGestureEnabled={false}
      removeClippedSubviews={true}
      renderItem={[Function]}
      scrollEnabled={true}
      scrollEventThrottle={1}
      scrollsToTop={false}
      shouldOptimizeUpdates={true}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      slideStyle={Object {}}
      sliderWidth={710}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 0,
          "width": 710,
        }
      }
      swipeThreshold={20}
      useScrollView={false}
      vertical={false}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={13}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            collapsable={false}
            nativeID="animatedComponent"
            pointerEvents="box-none"
            style={
              Object {
                "opacity": 1,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
                "width": 710,
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4280715245,
                  4291457515,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 10,
                    "marginTop": 10,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                  ]
                }
              >
                <View>
                  <Text
                    color="white"
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "SpoqaHanSansNeo-Bold",
                          "fontSize": 16,
                        },
                      ]
                    }
                    weight="bold"
                  >
                    나의 3대 측정
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "marginTop": 4,
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontSize": 16,
                          },
                          undefined,
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <Text
                      color="white"
                      size={12}
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontFamily": "SpoqaHanSansNeo-Regular",
                            "fontSize": 12,
                          },
                          Object {
                            "marginLeft": 4,
                          },
                        ]
                      }
                    >
                      21\`11.16
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "marginBottom": 40,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 20,
                    },
                  ]
                }
              >
                <Text
                  color="white"
                  style={
                    Array [
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "SpoqaHanSansNeo-Bold",
                        "fontSize": 16,
                      },
                    ]
                  }
                  weight="bold"
                >
                  로그인 한 후에 확인할 수 있어요!
                </Text>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#ffffff",
                      "borderRadius": 8,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "justifyContent": "center",
                      "marginTop": 10,
                      "opacity": 1,
                      "paddingBottom": 14,
                      "paddingLeft": 20,
                      "paddingRight": 20,
                      "paddingTop": 14,
                    }
                  }
                >
                  <Text
                    color="white"
                    size={12}
                    style={
                      Array [
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "SpoqaHanSansNeo-Bold",
                          "fontSize": 12,
                        },
                      ]
                    }
                    weight="bold"
                  >
                    로그인 하러가기
                  </Text>
                </View>
              </View>
            </BVLinearGradient>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "marginTop": 40,
          },
        ]
      }
    >
      <Text
        size={16}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "SpoqaHanSansNeo-Bold",
              "fontSize": 16,
            },
          ]
        }
        weight="bold"
      >
        오늘의 운동
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#ff190c",
              "borderRadius": 13,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 26,
              "justifyContent": "center",
              "marginRight": 10,
              "opacity": 1,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "width": 26,
            }
          }
        >
          <Text
            color="error"
            size={16}
            style={
              Array [
                Object {
                  "color": "#ff190c",
                  "fontFamily": "SpoqaHanSansNeo-Bold",
                  "fontSize": 16,
                },
              ]
            }
            weight="bold"
          >
            !
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "SpoqaHanSansNeo-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          오늘 계획한 운동이 없어요
          

          운동을 계획하러 가볼까요?
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2687ED",
              "borderRadius": 8,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginLeft": 10,
              "opacity": 1,
              "paddingBottom": 14,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 14,
            }
          }
        >
          <Text
            color="white"
            size={16}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "SpoqaHanSansNeo-Bold",
                  "fontSize": 16,
                },
              ]
            }
            weight="bold"
          >
            운동 계획하기
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
